docs/messaging/messaging-spans.md

```diff
@@ -512,6 +512,50 @@ flowchart LR;
 | `messaging.message.id` | `"a1"` | `"a2"` | |
 | `messaging.batch.message_count` |  |  | 2 |
 
+### Batch publishing
+
+Given is a publisher that publishes a batch with two messages to a topic "Q" on
+Kafka, and two different consumers receiving one of the messages.
+
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  direction TB
+  CA[Span Create A]
+  CB[Span Create B]
+  P[Span Publish]
+  end
+  subgraph CONSUMER1
+  direction TB
+  D1[Span Receive A]
+  end
+  subgraph CONSUMER2
+  direction TB
+  D2[Span Receive B]
+  end
+  CA-. link .-P;
+  CB-. link .-P;
+  CA-. link .-D1;
+  CB-. link .-D2;
+
+  classDef normal fill:green
+  class P,CA,CB,D1,D2 normal
+  linkStyle 0,1,2,3 color:green,stroke:green
+```
+
+| Field or Attribute | Span Publish | Span Create A | Span Create B | Span Receive A | Span Receive B |
```

Comment:
> * Changed the order so it's Create A / Create B / Publish / Receive A / Receive B, since that's the order they would be created in

Incorporated.

> * Made the publish span a Client SpanKind, added a Producer span kind for each create span

Done. In this scenario, "publish" can either be an internal or a client span.

> * Removed the parent spans and added bidirectional links between the publish span to each of the creates

Removed the parent spans. However, I didn't add bi-directional links. We should denote here which span is supposed to add the link.

> * Added bidrectional links from the create to receive spans

I didn't apply that, see the reason above.

> * Added the messaging.batch.message_count attribute

Done.